DQM/TrackingMonitorSource/plugins/ZtoMMMuonTrackProducer.cc

```diff
@@ -0,0 +1,127 @@
+
+#include "DataFormats/TrackReco/interface/HitPattern.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "TLorentzVector.h"
+
+#include "DQM/TrackingMonitorSource/interface/ZtoMMMuonTrackProducer.h"
+
+using namespace std;
+using namespace edm;
+//using namespace reco;
+
+ZtoMMMuonTrackProducer::ZtoMMMuonTrackProducer(const edm::ParameterSet& ps)
+    : muonTag_(ps.getUntrackedParameter<edm::InputTag>("muonInputTag", edm::InputTag("muons"))),
+      bsTag_(ps.getUntrackedParameter<edm::InputTag>("offlineBeamSpot", edm::InputTag("offlineBeamSpot"))),
+      muonToken_(consumes<reco::MuonCollection>(muonTag_)),
+      bsToken_(consumes<reco::BeamSpot>(bsTag_)),
+      maxEta_(ps.getUntrackedParameter<double>("maxEta", 2.4)),
+      minPt_(ps.getUntrackedParameter<double>("minPt", 5)),
+      maxNormChi2_(ps.getUntrackedParameter<double>("maxNormChi2", 0)),
+      maxD0_(ps.getUntrackedParameter<double>("maxD0", 0.02)),
+      maxDz_(ps.getUntrackedParameter<double>("maxDz", 20.)),
+      minPixelHits_(ps.getUntrackedParameter<uint32_t>("minPixelHits", 1)),
+      minStripHits_(ps.getUntrackedParameter<uint32_t>("minStripHits", 8)),
+      minChambers_(ps.getUntrackedParameter<uint32_t>("minChambers", 2)),
+      minMatches_(ps.getUntrackedParameter<uint32_t>("minMatches", 2)),
+      minMatchedStations_(ps.getUntrackedParameter<double>("minMatchedStations", 2)),
+      maxIso_(ps.getUntrackedParameter<double>("maxIso", 0.3)),
+      minPtHighest_(ps.getUntrackedParameter<double>("minPtHighest", 24)),
+      minInvMass_(ps.getUntrackedParameter<double>("minInvMass", 60)),
+      maxInvMass_(ps.getUntrackedParameter<double>("maxInvMass", 120)) {
+  produces<reco::TrackCollection>("");
+}
+
+ZtoMMMuonTrackProducer::~ZtoMMMuonTrackProducer() {}
+
+void ZtoMMMuonTrackProducer::produce(edm::Event& iEvent, edm::EventSetup const& iSetup) {
+  std::unique_ptr<reco::TrackCollection> outputTColl(new reco::TrackCollection());
+
+  // Read Muon Collection
+  edm::Handle<reco::MuonCollection> muonColl;
+  iEvent.getByToken(muonToken_, muonColl);
+
+  // and the beamspot
+  edm::Handle<reco::BeamSpot> beamSpot;
+  iEvent.getByToken(bsToken_, beamSpot);
+
+  std::vector<TLorentzVector> list;
+  if (muonColl.isValid()) {
+    for (auto const& mu : *muonColl) {
+      if (!mu.isGlobalMuon())
+        continue;
+      if (!mu.isPFMuon())
+        continue;
+      if (std::fabs(mu.eta()) >= maxEta_)
+        continue;
+      if (mu.pt() < minPt_)
+        continue;
+
+      reco::TrackRef gtk = mu.globalTrack();
+      double chi2 = gtk->chi2();
+      double ndof = gtk->ndof();
+      double chbyndof = (ndof > 0) ? chi2 / ndof : 0;
+      if (chbyndof >= maxNormChi2_)
+        continue;
+
+      reco::TrackRef tk = mu.innerTrack();
+      double trkd0 = tk->d0();
+      double trkdz = tk->dz();
```

Comment:
```suggestion
```
No need to define, and assign, these two variables outside of the if scope below